cmd/google_cloud_ops_agent_engine/main.go

```diff
@@ -44,5 +45,27 @@ func run() error {
 	if err := confgenerator.MergeConfFiles(*input, confDebugFolder, "linux", apps.BuiltInConfStructs); err != nil {
 		return err
 	}
+	if err := logConfigFiles(filepath.Join(confDebugFolder, "built-in-config.yaml"), filepath.Join(confDebugFolder, "merged-config.yaml")); err != nil {
+		return err
+	}
 	return confgenerator.GenerateFiles(filepath.Join(confDebugFolder, "merged-config.yaml"), *service, *logsDir, *stateDir, *outDir)
 }
+
+// logConfigFiles logs the built-in and merged config files to STDOUT. These are then written by journald to var/log/syslog and so to
+// Cloud Logging once the ops-agent is running.
+func logConfigFiles(builtInConfigFile, mergedConfigFile string) error {
```

Comment:
I wonder if it would make more sense to do this within `mergeConfFiles`. Maybe see what @quentinmit thinks.

Reply:
I thought about that but since this is logged differently for Linux (stdout -> journald -> logging agent) and Windows (event log -> logging agent), I thought it probably should live outside the common code

Reply:
My main comment here is that we shouldn't be reading the files to log them; we should be serializing the in-memory config directly out to the log. I believe we have plans to get rid of the on-disk merged config file at some point.

I don't particularly care about formatting; we can clean that up as part of the improved/structured logging work that @davidbtucker is embarking on.

Reply:
Restructured some of the changes so we don't rely on reading from files and instead use the in memory structures. Additionally, I moved some code around so we don't rely on reading from these files unnecessarily.

These files are now written to just for debugging purposes.